### ISSUE TITLE
Validator - an error in debug tools, regarding a field sticked to a non-validated group

### ISSUE THREAD

**Issue by konrados** [symfony/symfony#17310]
Hi,

These are my first days with Symfony, so I'm not really sure if this is a bug or not. But I believe it is.

I have an entity "User" and a variable:

    /**
    * @Assert\NotBlank
    * @Assert\Length(max = 4096)
    */
	private $plainPassword;

As you can see it doesn't belong to any group.

Now, in a controller I have this:

    $user = new \AppBundle\Entity\User();
    $form = $this->createForm(Register::class, $user);
    $form->handleRequest($request);

    if ($form->isSubmitted()) {
        $validator = $this->get('validator');
        $errors = $validator->validate($user, null, array('registration'));
        dump($errors);
        if(count($errors) == 0){
            return $this->redirectToRoute('user_register',array('done' => 1));
            }
    }
    return $this->render('front/users/register.html.twig', [
        'form'=>$form->createView(),
    ]);
    
	

Notice that I want to validate on "registration" group only.

And finally the register class building a form:

    class Register extends AbstractType
    {
        public function buildForm(FormBuilderInterface $builder, array $options)
        {
            $builder
                ->add('username')
                ->add('email',  \Symfony\Component\Form\Extension\Core\Type\TextType::class, array('error_bubbling'=>true))
                ->add('password',PasswordType::class)
                ->add('confirmPassword',PasswordType::class)    
                ->add('save', SubmitType::class);
        }
    }

Notice there is no "plainPassword" field there.

Now, after submitting a form and clicking on Web debug toolbar, I can see that there is an error
in a form - when I click it, I can see that the error is related to the plainPassword var. Isn't that true, that this field should not be validated, since it doesn't belong to the "registration" group?

However, the $errors variable (returned by validate function) is empty. That is count($errors) == 0; So theoretically it's fine, I do the redirection etc.

However2 - if I won't redirect and display the form again, an error is displayed at the top of the form: "This value should not be blank".

Update: I forgot to mention that when I explicitly assign the variable to a group, e.g. "test_group" then everything's fine, but from the docs it looks like not assigning should automatically assign to a "default" group.

p.s. I don't know how to add a label to an issue, or maybe I don't have rights to do this.

**Comment by stof** [comment]
@konrados the error shown in the WDT comes from the validation performed in the form, not from the validation triggered by your own code (which is not tracked by the WDT). And your form is still running validation in the default group, not in the registration group.

**Comment by stof** [comment]
and please use the [support channels](http://symfony.com/support) to ask for support. The github issue tracker is meant to track bugs and feature requests. It is not a support forum (which would make it impossible to manage).
so I'm closing this issue

**Comment by konrados** [comment]
Hi, thanks for answering.

I was pretty sure it's bug, this is why I added this issue. It wasn't supposed to be a question.

I know you guys prefer StackOverflow or irc as the support channels, this was  irc where I was suggested to file an issue.

And yes, I started with StackOverlow ( http://stackoverflow.com/questions/34673472/custom-validaton-messages-not-working-on-symfony-3 ) where I did not get much help :( (This is a different question, but this question started my investigation which led me here).

Anyway, I'm sorry for wasting your time, but again, I was sure it was a bug.
